libs/Microsoft.MixedReality.WebRTC.Native/src/peer_connection.cpp

```diff
@@ -1678,39 +1675,31 @@ PeerConnectionImpl::GetOrCreateTransceiverForNewRemoteTrack(
 }
 
 void PeerConnectionImpl::SynchronizeTransceiversUnifiedPlan(bool remote) {
-  auto rtp_transceivers = peer_->GetTransceivers();
-  const int num_transceivers = static_cast<int>(rtp_transceivers.size());
-  for (int mline_index = 0; mline_index < num_transceivers; ++mline_index) {
-    // RTP transceivers are in mline_index order by design
-    const rtc::scoped_refptr<webrtc::RtpTransceiverInterface>& tr =
-        rtp_transceivers[mline_index];
-    RefPtr<Transceiver> transceiver;
-    {
-      rtc::CritScope lock(&transceivers_mutex_);
-      if (mline_index < transceivers_.size()) {
-        transceiver = transceivers_[mline_index];
-      }
-      RTC_DCHECK(!transceiver || (transceiver->impl() == tr));
-    }
+  for (auto&& rtp_tr : peer_->GetTransceivers()) {
+    const int mline_index = ExtractMlineIndexFromRtpTransceiver(rtp_tr);
+    RefPtr<Transceiver> transceiver = FindWrapperFromRtpTransceiver(rtp_tr);
```

Comment:
This is a quadratic loop. The element of `peer_->GetTransceivers()` and `transceivers_` should match by index, right? If not it should be easy enough to match them linearly anyway. I don't expect this to be a problem in practice but better safe than sorry 😄

Reply:
Fixed with sorting by memory address of RTP transceiver. The index probably match, but this was an assumption about an implementation detail that I am glad to depart from. This should have no impact on performance in any case, as we expect no more than a few transceivers.